### PR TITLE
Improved Preferences instantiation

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Purpose / Description
Initially, it was a `KotlinCleanup` task, but going through the file, I thought we can have a better implementation of it. 

## Fixes
Improves code quality

## Approach
Using JUnit's `@Before` method to create a new instance of `Preferences` rather than creating it each time manually.

## How Has This Been Tested?
Tests run fine, and app works as expected on Realme 6i API 30

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt [hunk 1]
```diff
@@ -18,17 +18,24 @@ package com.ichi2.anki
...
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class PreferencesTest : RobolectricTest() {
+    private lateinit var preferences: Preferences
+
+    @Before
+    fun setup() {
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt:32]: you want to override the `@Setup` method in `RobolectricTest`

**Review by david-allison — COMMENTED** [review]
Seems good, one minor improvement to ensure correctness

**Review by david-allison — CHANGES_REQUESTED** [review]
Conflicts also need fixing - marking as 'changes needed'

File: AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt [hunk 2]
```diff
@@ -28,8 +28,8 @@ import org.junit.runner.RunWith
 class PreferencesTest : RobolectricTest() {
     private lateinit var preferences: Preferences
 
-    @Before
-    fun setup() {
+    override fun setUp() {
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt:30]: And `@Before` (I beleive)

Comment (divyansh-dxn) [reply AnkiDroid/src/test/java/com/ichi2/anki/PreferencesTest.kt:30]: I don't know whether annotations get inherited or not, but still, in other tests, it has been written again in the subclass, so I'm writing it.

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
excellent - thank you!

**Comment by mikehardy** [comment]
Hi there! APRIL OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for [April 2022 submissions](https://github.com/ankidroid/Anki-Android/pulls?q=is%3Apr+merged%3A2022-04-01..2022-04-30)

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of March.

Please note that GSoC contributions are okay for this process. Our philosophy is that our users have donated to AnkiDroid for all contributions. The only PRs that will not go through the OpenCollective process are ones directly related to am accepted GSoC project from a selected participant, since those receive a stipend from GSoC itself.

Please understand that this means the monthly budget we have for payments might be stretched a bit and the cap on payments-per-person may be lower, but we try to make our process as fair and transparent as possible, we just need your understanding.

Thanks!

**Pull request merged**